### PR TITLE
4.1.x: Support Java 17 & Dependency Updates

### PR DESCRIPTION
**Pull request by volosied**

### CODE REVIEW

**Comment by volosied** [comment]
@melloware @tandraschko  I'm getting the following error.  Could either of figure out why? 

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.myfaces.core.extensions.quarkus.showcase.QuarkusMyFacesShowcaseTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.393 s <<< FAILURE! - in org.apache.myfaces.core.extensions.quarkus.showcase.QuarkusMyFacesShowcaseTest
[ERROR] org.apache.myfaces.core.extensions.quarkus.showcase.QuarkusMyFacesShowcaseTest  Time elapsed: 1.393 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.apache.myfaces.core.extensions.quarkus.deployment.MyFacesProcessor
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.apache.myfaces.core.extensions.quarkus.deployment.MyFacesProcessor
Caused by: java.lang.ClassNotFoundException: org.apache.myfaces.core.extensions.quarkus.deployment.MyFacesProcessor

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   QuarkusMyFacesShowcaseTest » Runtime java.lang.RuntimeException: java.lang.Cla...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

File: parent/pom.xml [hunk 1]
```diff
@@ -510,7 +507,7 @@
     </distributionManagement>
 
     <properties>
-        <java.version>11</java.version>
+        <java.version>17</java.version>
```
Comment (melloware) [inline parent/pom.xml:510]: Will this make 4.1 JDK 17 only for users?

Comment (volosied) [reply parent/pom.xml:510]: It would.

Comment (volosied) [reply parent/pom.xml:510]: Do you need Quarkus on Java 11?

Comment (melloware) [reply parent/pom.xml:510]: Nope i was just making sure 4.1 minimum version would be 17 now.  I know 4.0 we are keeping at 11 right?

File: extensions/quarkus/showcase/pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
         <surefire.version>2.22.0</surefire.version>
 
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>11</java.version>
+        <java.version>17</java.version>
```
Comment (volosied) [inline extensions/quarkus/showcase/pom.xml:37]: Reverting this change make the tests pass: 
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache MyFaces Core 4.1 - Extensions 4.1.0-SNAPSHOT:
[INFO] 
[INFO] Apache MyFaces Core 4.1 - Extensions ............... SUCCESS [  1.078 s]
[INFO] Apache MyFaces Core 4.1 - Extensions - Quarkus ..... SUCCESS [  0.201 s]
[INFO] Apache MyFaces Core 4.1 - Extensions - Quarkus - Runtime SUCCESS [  2.328 s]
[INFO] Apache MyFaces Core 4.1 - Extensions - Quarkus - Deployment SUCCESS [  1.365 s]
[INFO] Apache MyFaces Core 4.1 - Extensions - Quarkus - Showcase SUCCESS [  9.112 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14.224 s
[INFO] Finished at: 2024-01-22T15:23:22-05:00
[INFO] ------------------------------------------------------------------------
```

@melloware

Comment (melloware) [reply extensions/quarkus/showcase/pom.xml:37]: LGTM!

File: parent/pom.xml (hunk 1 above)
Comment (volosied) [reply parent/pom.xml:510]: Correct -- 4.0 is still 11.   4.1 is now part of EE11 which requires a minimum of Java 17.

File: extensions/quarkus/showcase/pom.xml (hunk 2 above)
Comment (volosied) [reply extensions/quarkus/showcase/pom.xml:37]: Hmm -- I tried again and it failed?  :(  Not sure what's going on....

Once I get past this, I think I can get some releases out

Comment (melloware) [reply extensions/quarkus/showcase/pom.xml:37]: let me take a look

**Comment by melloware** [comment]
In the Quarkus Showcase pom.xml shouldn't this be changed to Jakarta?

```xml
<dependency>
            <groupId>javax.validation</groupId>
            <artifactId>validation-api</artifactId>
            <version>2.0.1.Final</version>
        </dependency>
```

**Comment by melloware** [comment]
I just removed it i don't even think its needed.

**Comment by melloware** [comment]
I just built your branch locally and its working for me now.

image

**Comment by melloware** [comment]
switching to 11 in Showcase pom.xml still works for me.

**Comment by volosied** [comment]
It looks to work now with 17 ( new terminal).  I guess something got messed up. Let me clean up the PR and we can merge

**Comment by melloware** [comment]
Oh yeah I tested 17 first. All working once I removed that `javax.validation` depenendency which was not needed.

**Review by melloware — APPROVED** [review]

**Pull request merged**